File: SkaiLab.Invoice/ClientApp/src/app/models/employee.ts
import { Country } from "./country";
import { Currency } from "./currency";
import { Gender } from "./gender";
import { MaritalStatus } from "./marital-status";
import { SalaryType } from "./salary-type";

export class Employee{
    id:number;
    firstName:string;
    lastName:string;
    displayName:string;
    dateOfBirth?:Date;
    genderId:number;
    gender:Gender;
    phoneNumber:string;
    email:string;
    address:string;
    maritalStatusId:number;
    maritalStatus:MaritalStatus;
    salaryTypeId:number;
    salaryType:SalaryType;
    isResidentEmployee:boolean;
    iDOrPassportNumber:string;
    documentUrl:string;
    organisationId:string;
    salary:number;
    numberOfChild:number;
    currencyId:number;
    currency:Currency;
    isActive:boolean;
    jobTitle:string;
    countryId:number;
    country:Country;
    isConfederationThatHosts:boolean;
    constructor(){
        this.id=0;
        this.firstName="";
        this.lastName="";
        this.displayName="";
        this.dateOfBirth=null;
        this.genderId=null;
        this.gender=new Gender();
        this.phoneNumber="";
        this.email="";
        this.address="";
        this.maritalStatusId=null;
        this.maritalStatus=new MaritalStatus();
        this.salaryTypeId=null;
        this.salaryType=new SalaryType();
        this.isResidentEmployee=true;
        this.iDOrPassportNumber="";
        this.documentUrl="";
        this.organisationId="";
        this.salary=0;
        this.numberOfChild=0;
        this.currencyId=0;
        this.currency=new Currency();
        this.isActive=true;
        this.jobTitle="";
        this.countryId=39;
        this.isConfederationThatHosts=false;
       
    }
}

File: SkaiLab.Invoice.Models/Employee.cs
using System;
namespace SkaiLab.Invoice.Models
{
    public class Employee
    {
        public long Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string DisplayName { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public int GenderId { get; set; }
        public Gender Gender { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public int MaritalStatusId { get; set; }
        public MaritalStatus MaritalStatus { get; set; }
        public int SalaryTypeId { get; set; }
        public SalaryType SalaryType { get; set; }
        public bool IsResidentEmployee { get; set; }
        public string IDOrPassportNumber { get; set; }
        public string DocumentUrl { get; set; }
        public string OrganisationId { get; set; }
        public decimal Salary { get; set; }
        public int NumberOfChild { get; set; }
        public int CurrencyId { get; set; }
        public Currency Currency { get; set; }
        public bool IsActive { get; set; }
        public string JobTitle { get; set; }
        public int CountryId { get; set; }
        public Country Country { get; set; }
        public bool IsConfederationThatHosts { get; set; }
    }
}
